packages/core/src/particle/ParticleGenerator.ts

```diff
@@ -764,4 +766,150 @@ export class ParticleGenerator {
+    // Transform gravity direction into local space
+    direction.copyFrom(this._renderer.scene.physics.gravity);
+    Vector3.transformByQuat(direction, worldInvQuat, direction);
+
+    this._adjustBoundingBoxFromDirectionalVelocity(
+      direction,
+      direction,
+      0.5 * lifetime * lifetime,
+      this.main.gravityModifier
+    );
+
+    // StartSize's impact
+    const maxSize = this.main.startSize3D
+      ? Math.max(
+          this.main.startSizeX._getMinMaxValue({ min: 0, max: 0 }).max,
+          this.main.startSizeY._getMinMaxValue({ min: 0, max: 0 }).max,
+          this.main.startSizeZ._getMinMaxValue({ min: 0, max: 0 }).max
+        )
+      : this.main.startSize._getMinMaxValue({ min: 0, max: 0 }).max;
+
+    min.x -= maxSize;
+    max.x += maxSize;
+
+    min.y -= maxSize;
+    max.y += maxSize;
+
+    min.z -= maxSize;
+    max.z += maxSize;
+
+    // VelocityOverLifetime Module's impact
+    if (this.velocityOverLifetime.enabled) {
+      const { velocityX, velocityY, velocityZ } = this.velocityOverLifetime;
+      directionMin.set(
+        velocityX._getMinMaxValue({ min: 0, max: 0 }).min,
+        velocityY._getMinMaxValue({ min: 0, max: 0 }).min,
+        velocityZ._getMinMaxValue({ min: 0, max: 0 }).min
+      );
+      directionMax.set(
+        velocityX._getMinMaxValue({ min: 0, max: 0 }).max,
+        velocityY._getMinMaxValue({ min: 0, max: 0 }).max,
+        velocityZ._getMinMaxValue({ min: 0, max: 0 }).max
+      );
+
+      if ((this.velocityOverLifetime.space = ParticleSimulationSpace.World)) {
+        Vector3.transformByQuat(directionMin, worldInvQuat, directionMin);
+        Vector3.transformByQuat(directionMax, worldInvQuat, directionMax);
+      }
+
+      this._adjustBoundingBoxFromDirectionalVelocity(directionMin, directionMax, lifetime);
+    }
+
+    // SimulationSpace's Impact
+    if (this.main.simulationSpace === ParticleSimulationSpace.World) {
+      const worldPosition = this._renderer.entity.transform.worldPosition;
+      min.x -= Math.max(0, worldPosition.x);
+      max.x -= Math.min(0, worldPosition.x);
+
+      min.y -= Math.max(0, worldPosition.y);
+      max.y -= Math.min(0, worldPosition.y);
+
+      min.z -= Math.max(0, worldPosition.z);
+      max.z -= Math.min(0, worldPosition.z);
+    }
+  }
+
+  private _adjustBoundingBoxFromDirectionalVelocity(
+    directionMax: Vector3,
+    directionMin: Vector3,
+    factor: number,
+    velocity?: ParticleCompositeCurve
+  ): void {
+    const { min, max } = this._renderer._localBounds;
+
+    const val = { max: 0, min: 0 };
+    velocity._getMinMaxValue(val);
+
+    const velocityMin = velocity ? factor * val.min : factor;
+    const velocityMax = velocity ? factor * val.max : factor;
+
+    min.x += Math.min(
+      0,
+      directionMin.x * velocityMin,
+      directionMin.x * velocityMax,
+      directionMax.x * velocityMin,
+      directionMax.x * velocityMax
```

Comment:
`directionMin.x * velocityMax` is not smallest?

Reply:
directionMax.x * velocityMin could be negative and the smallest